### PR TITLE
setup protractor-browser-logs on onPrepare

### PR DESCRIPTION
**Pull request by ronami**
will fix https://github.com/wix-private/wix-js-stack/issues/521

### CODE REVIEW

File: yoshi/test/test.spec.js [hunk 1]
```diff
@@ -94,6 +94,43 @@ describe('Aggregator: Test', () => {
       expect(res.stdout).to.contain('1 spec, 0 failures');
     });
 
+    it(`use protractor-browser-logs and fail if there are any console errors on the browser`, () => {
```
Comment (yanivefraim) [inline yoshi/test/test.spec.js]: we're missing a test for same use case but with FT turned off. @yardnsm - can you plz help with that?

File: yoshi/config/protractor.conf.js [hunk 2]
```diff
@@ -62,6 +64,24 @@ const merged = ld.mergeWith({
     });
   },
   onPrepare: () => {
+    if (shouldUseProtractorBrowserLogs) {
+      const browserLogs = require('protractor-browser-logs');
```
Comment (roymiloh) [inline yoshi/config/protractor.conf.js]: Please move this block to another function.

File: yoshi/package.json [hunk 3]
```diff
@@ -71,6 +71,7 @@
     "phantomjs-polyfill": "~0.0.2",
     "postcss-loader": "~2.0.6",
     "protractor": "~5.1.2",
+    "protractor-browser-logs": "^1.0.351",
```
Comment (roymiloh) [inline yoshi/package.json]: + managed deps.

File: yoshi/test/test.spec.js [hunk 4]
```diff
@@ -94,6 +94,78 @@ describe('Aggregator: Test', () => {
       expect(res.stdout).to.contain('1 spec, 0 failures');
     });
 
+    it(`should use protractor-browser-logs and fail if there are any console errors on the browser`, () => {
+      const res = test
+        .setup({
+          'protractor.conf.js': `
+            const http = require("http");
```
Comment (roymiloh) [inline yoshi/test/test.spec.js]: Can't we use a template (from fixtures)?

**Comment by lbelinsk** [comment]
@yardnsm 
Although this is already merged, I think we should support custom configurations from the users. For example, to ignore some specific error message. Basically to allow the users to extend yoshi's base browser logs..

**Comment by roymiloh** [comment]
@lbelinsk We expose [its instance](https://github.com/wix/yoshi/pull/202/files#diff-900eb771d6de633707018f52e16f7eefR105), so you can configure that as you wish.

**Pull request merged**